pkg/ruler/frontend_decoder.go

```diff
@@ -18,8 +20,18 @@ const (
 )
 
 type JsonDecoder struct{}
+type ProtobufDecoder struct{}
 type Warning []string
```

Comment:
Warnings?

Reply:
What do you mean is we should return nil at the `ProtobufDecoder ` decode func since the `error` is only decoded by the json codec?

Reply:
No... Just the naming thing. Better to call it `Warnings` instead of `Warning`

Reply:
Thanks, fixed it.